src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -1142,8 +1139,7 @@ private Binder GetParameterNameAttributeValueBinder(MemberDeclarationSyntax memb
                     return new WithParametersBinder(method.Parameters, nextBinder);
                 }
 
-                if (memberSyntax is RecordDeclarationSyntax { ParameterList: { ParameterCount: > 0 } }
-                    or RecordStructDeclarationSyntax { ParameterList: { ParameterCount: > 0 } })
+                if (memberSyntax is RecordDeclarationSyntax { ParameterList: { ParameterCount: > 0 } })
```

Comment:
>RecordDeclarationSyntax [](http://example.com/codeflow?start=36&length=23)

Have you confirmed that all remaining type checks for RecordDeclarationSyntax are applicable to record structs or appropriately adjusted? Same for Visit methods.

Reply:
Yes. I have a list. The places that previously only handled RecordDeclarationSyntax are guarded (kind check or base type check). Some are guarded locally and are checked elsewhere (unreachable).
For the latter cases, I didn't see anything wrong with them. I could add assertions if you think that's useful.
An example is `ExpressionVariableFinder.VisitRecordDeclaration`.

---
In reply to: [616728346](https://github.com/dotnet/roslyn/pull/52702#discussion_r616728346) [](ancestors = 616728346)